tools/code_format/check_format.py

```diff
@@ -552,92 +542,8 @@ def check_runtime_flags(self, file_path, error_messages):
                     error_messages.append("%s and %s are out of order\n" % (line, previous_flag))
             previous_flag = line
 
-    def check_current_release_notes(self, file_path, error_messages):
```

Comment:
@phlax can you get the person who wrote this code to review the changes? I don't have enough familiarity here.

Reply:
in terms of maintainers - the last person to do significant work on this file in the related lines, other than linting was @alyssawilk  - are you up for reviewing ?

it moves the logic out of here to a python class - and adds unit tests - with 100% coverage - so im pretty confident in the code

Reply:
OK, as discussed offline, happy to review with the understanding I'm basically just doing a syntax pass and semantics are preserved.